### DISCUSSION TITLE
"Call node event failed because of invalid path", regardless of what path is written

### DISCUSSION THREAD

**Discussion by aumusic** [dialogic-godot/dialogic#1752] [Q&A] [answered]
I don't know whether the problem is me on this one (though, for the sake of the discussion, I'm going to assume it is), but is it possible in any way to get an explanation of just what is required for a node path when using the "call node" parameter? I've tried multiple approaches (res://etc, /root/node2D/etc, assigning an Autoload name to the node and listing that), and every one of them gives me the same "call node event failed because of invalid path" error. 

Thanks in advance.

**Comment by 0Volcanon** [comment]
Make sure the names are entered correctly. The methods must be only the name, "**method_name**" instead of "**method_name()**" with arguments passed in the block settings. The path can best be copied by running the game, switching to the Remote tree, right clicking the desired node and "Copy Node Path." You can also use the node name relative to the current scene.

**Reply by aumusic** [reply]
Thanks for this. Apparently I had something wrong somewhere (though I never did get the paths working correctly, though eventually I was able to get the autoload names to work).

**Comment by RadicalRobbie** [comment]
I'm also having this problem. I did exactly as 0Volcanon suggested and it's giving me the same error
![image](https://github.com/coppolaemilio/dialogic/assets/145951315/bc9eabeb-3e76-49fe-a856-9461ed7d2834)

**Reply by aumusic** [reply]
@RadicalRobbie , as I noted before, I didn't get the locations working for me, but I ended up deleting all of the calls within my Dialogic script, as well as the Autoload listings, and re-creating all of them (and using Autoload names within the re-written calls for the object names), and that worked (which leads me to believe I had something screwed up somewhere along the line). I hope that works for you.

**Reply by 0Volcanon** [reply]
I now see the issue. Apparently using "Copy Node Path" doesn't return the full path. Also, my comment regarding node name relative to active scene was also incorrect. You need to change the node path from "MeshInstance3D" to "root/Node3D/MeshInstance3D" (if you add no '/' before root, Dialogic will add it by default) and it should work. Sorry for the misunderstanding. Hopefully, this fixes the issue.
Edit: You can also skip "/root" as well since Dialogic adds that in as well. So, @aumusic must have had some issue in the paths (perhaps a misspelling) or if it still doesn't work, try getting the addon from the master branch.

**Comment by Jowan-Spooner** [comment, marked as answer]
Okay just to have a finite answe here:

Using paths in the Call Node event is tricky. You can either
- use an autoload name (make sure spelling and capitalization are correct!)
- use an absolute path starting from the **scene trees ROOT node**, which is different from your scenes root node (when running your game you can switch to the "remote tree" instead of the local tree and right click a node there to get the absolute path)

**Reply by RadicalRobbie** [reply]
Can verify that this worked after a little messing around. 
![image](https://github.com/coppolaemilio/dialogic/assets/145951315/86374eba-ebfe-40a1-bddd-733eacdf9361)
![image](https://github.com/coppolaemilio/dialogic/assets/145951315/ac596ffd-73a0-4cf4-99c4-0f62e2452e49)
![image](https://github.com/coppolaemilio/dialogic/assets/145951315/70d202b7-c461-412e-b658-26ae946d9853)
![image](https://github.com/coppolaemilio/dialogic/assets/145951315/cece4a36-2581-44b5-ae96-83cdfff52c2c)
